[WIP] Feat: 유저 auth 기능 추가

anewjean opened the pull request.

anewjean committed:
[WIP] Feat: 유저 auth 기능 추가

changchanghwang commented on .gitignore:
```diff
@@ -11,7 +11,7 @@
 
 # IntelliJ
-.idea
+../.idea
```
Why changed?

anewjean replied:
수정 전으로 변경했습니다

changchanghwang commented on src/main/kotlin/com/example/antserver/domain.user/AuthService.kt:
```diff
@@ -0,0 +1,167 @@
+            )
+
+            val googleUser = googleUserInfoResponse.body
+                ?: throw IllegalStateException("Google에서 사용자 정보를 가져올 수 없습니다.")
```
이런건 error http status code가 어떻게되나요?

anewjean replied:
400(BAD_REQUEST) 입니다

changchanghwang commented on src/main/kotlin/com/example/antserver/domain.user/AuthService.kt:
```diff
@@ -0,0 +1,167 @@
+            )
+
+            val googleUser = googleUserInfoResponse.body
+                ?: throw IllegalStateException("Google에서 사용자 정보를 가져올 수 없습니다.")
```
이런건 error http status code가 어떻게되나요?

anewjean replied:
400(BAD_REQUEST) 입니다

changchanghwang commented on src/main/kotlin/com/example/antserver/domain.user/AuthService.kt:
```diff
@@ -0,0 +1,167 @@
+                user = userRepository.save(user)
+            }
+            user
+        } catch (e: Exception) {
```
Try catch 안쓰고 글로벌하게 잡을수없나요?

anewjean replied:
GlobalExceptionHandler 추가했는데 아직 try catch 다 못 지웠습니다

changchanghwang commented on src/main/kotlin/com/example/antserver/domain.user/AuthService.kt:
```diff
@@ -0,0 +1,167 @@
+            }
+            user
+        } catch (e: Exception) {
+            throw IllegalStateException("인증 실패: ${e.message}")
```
이렇게 던지면 저한테 오는 에러메세지는 어떤건가요?

anewjean replied:
음...;; 수정할게요

changchanghwang commented on src/main/kotlin/com/example/antserver/domain.user/AuthService.kt:
```diff
@@ -0,0 +1,167 @@
+        }
+    }
+
+    fun setTokenToHeader() {
```
이건 왜필요한가요?

anewjean replied:
확인 후 지웠습니다

changchanghwang commented on src/main/kotlin/com/example/antserver/domain.user/RefreshToken.kt:
어그리거트로 쓸거면 패키지 분리부탁드립니다.

anewjean replied:
변경완료했습니다

changchanghwang commented on src/main/kotlin/com/example/antserver/domain.user/User.kt:
```diff
@@ -0,0 +1,46 @@
+import java.util.UUID
+
+data class User(
+    val id: UUID = UUID.randomUUID(),
```
Why use v4?

anewjean replied:
v7로 변경했습니다

changchanghwang commented on src/main/kotlin/com/example/antserver/domain.user/UserRoleType.kt:
```diff
@@ -0,0 +1,7 @@
+package domain.user
+
+enum class UserRoleType {
+    GUEST,
```
guest도 만들건가요?

anewjean replied:
아 빼야하는데 못 뺐어요

changchanghwang commented on src/main/kotlin/com/example/antserver/presentation/UserController.kt:
```diff
@@ -0,0 +1,60 @@
+    fun authenticate(
+        @RequestBody authRequest: UserAuthRequest
+        ): ResponseEntity<CommonResponse<UserAuthResponse>> {
+            val authorizationCode = authRequest.authorizationCode
+            val provider = authRequest.provider
```
```suggestion
            val (authorizationCode,provider) = authRequest
```

anewjean replied:
반영했습니다

changchanghwang commented on src/main/kotlin/com/example/antserver/presentation/UserController.kt:
```diff
@@ -0,0 +1,60 @@
+
+            val authResult = userService.authenticateUser(authorizationCode, provider)
+            val headers = authResult.headers
+            return ResponseEntity.ok()
+                .headers(headers)
+                .body(
+                    CommonResponse.success(
+                        data = UserAuthResponse(authResult.userName, authResult.userEmail, authResult.userRole)
+                    )
+                )
```
<img width="328" alt="image" src="https://github.com/user-attachments/assets/fc542896-c106-43d5-b5d1-ed4f7fe11f57">

accessToken만 있으면 됩니다.

anewjean replied:
최종적으로 userId claim해서 만든 access token과 refresh token 반환합니다

changchanghwang commented on src/main/kotlin/com/example/antserver/presentation/dto/GoogleAccessTokenRequest.kt:
```diff
@@ -0,0 +1,13 @@
+
+//@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy::class)
+data class GoogleAccessTokenRequest(
+    val clientId: String,
+    val clientSecret: String,
+    val redirectUri: String,
+    val responseType: String,
+    val scope: String,
+    val code: String,
+    val accessType: String,
+    val grantType: String,
```
<img width="715" alt="image" src="https://github.com/user-attachments/assets/a186c5a0-92d1-4aa6-819b-f8de4f51fca3">
얘네만 있으면 되는거 아니에여?

anewjean replied:
맞습니다

changchanghwang commented on src/main/kotlin/com/example/antserver/presentation/dto/GoogleAccessTokenResponse.kt:
```diff
@@ -0,0 +1,11 @@
+//@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy::class)
+data class GoogleAccessTokenResponse(
+    val accessToken: String,
+    val expiresIn: String,
+    val refreshToken: String,
+    val scope: String,
+    val tokenType: String,
+    val idToken: String
```
왜 필요한가요?

anewjean replied:
idToken만 남겼습니다

changchanghwang commented on src/main/kotlin/com/example/antserver/presentation/dto/GoogleUserResponse.kt:
```diff
@@ -0,0 +1,25 @@
+
+//@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy::class)
+data class GoogleUserResponse(
+    val iss: String,
+    val azp: String,
+    val aud: String,
+    val sub: String,
+    val email: String,
+    val emailVerified: String,
+    val atHash: String,
+    val name: String,
+    val picture: String,
+    val givenName: String,
+    val familyName: String,
+    val locale: String,
+    val iat: String,
+    val exp: String,
+    val alg: String,
+    val kid: String,
+    val typ: String
```
필요한 애들만 가져와주세요

anewjean replied:
sub(사용자 식별값 -> user.providerId), email, emailVerified, name만 남겼습니다

changchanghwang commented on src/main/kotlin/com/example/antserver/presentation/dto/user/UserAuthRequest.kt:
```diff
@@ -0,0 +1,10 @@
+    val authorizationCode: String,
+    val provider: ProviderType
+) {
+    companion object
```
why need?

anewjean replied:
지웠습니다

changchanghwang commented on src/main/kotlin/com/example/antserver/util/BaseEntity.kt:
```diff
@@ -0,0 +1,24 @@
+
+@MappedSuperclass
+@EntityListeners(AuditingEntityListener::class)
+abstract class BaseEntity {
```
Aggregate로 naming하는건 어떤가요? or AggregateRoot

왜냐면 어차피 하위 어그리거트들은 루트어그리거트에 종속되어 life cycle이 정해지기때문에 상속할 필요가 없습니다.

changchanghwang commented on src/main/kotlin/com/example/antserver/util/filter/CustomAccessDeniedHandler.kt:
```diff
@@ -0,0 +1,19 @@
+        response: HttpServletResponse?,
+        accessDeniedException: AccessDeniedException?
+    ) {
+        response?.sendError(HttpServletResponse.SC_FORBIDDEN) //403
```
no need comment

anewjean replied:
지웠습니다

changchanghwang commented on src/main/kotlin/com/example/antserver/util/filter/CustomAuthenticationEntryPoint.kt:
```diff
@@ -0,0 +1,18 @@
+        response: HttpServletResponse?,
+        authException: AuthenticationException?
+    ) {
+        response?.sendError(HttpServletResponse.SC_UNAUTHORIZED) // 401
```
no need comment

anewjean replied:
지웠습니다

changchanghwang commented on src/main/kotlin/com/example/antserver/util/filter/JwtAuthenticationFilter.kt:
```diff
@@ -0,0 +1,92 @@
+                ?.let { userId ->
+                    userRepository.findById(userId)?.let { user ->
+                        if (user.role == UserRoleType.GUEST) {
+                            user.updateRole(UserRoleType.MEMBER) // 역할 변경
+                            userRepository.save(user) // 변경 내용 저장
```
no need comment

anewjean replied:
지웠습니다

changchanghwang commented on src/main/kotlin/com/example/antserver/util/filter/JwtAuthenticationFilter.kt:
```diff
@@ -0,0 +1,92 @@
+    ) {
+        val accessToken = authService.extractAccessToken(request)
+
+        // Access Token이 유효한 경우 사용자 인증
```
no need comment

anewjean replied:
지웠습니다

changchanghwang commented on src/main/kotlin/com/example/antserver/util/response/CommonResponse.kt:
```diff
@@ -0,0 +1,20 @@
+package com.example.antserver.util.response
+
+data class CommonResponse<T>(
+    val data: T? = null,
+    val errorMessage: String? = null
```
음? data에 error message넣는거 아니었어요?

anewjean replied:
아 이거 말씀드리려고 했는데 
    @PostMapping("/auth")
    fun authenticate(
        @RequestBody authRequest: UserAuthRequest
        ): ResponseEntity<CommonResponse<**UserAuthResponse**>> 
        
이런 식으로 data에 대한 타입 추론해서 message String 타입 들어가면 처리가  안 돼요
        
ResponseEntity<CommonResponse<**Any**>> 로 써도 되긴 하는데 Any 쓰고싶지 않아서 일단은 저렇게 해뒀습니다

anewjean replied:
errorMessage 삭제했습니다

changchanghwang committed:
Merge branch 'main' into auth

anewjean committed:
[WIP] Fix: auth 로직 수정 및 auth 이외 불필요한 로직 주석처리

changchanghwang commented on .gitignore:
```diff
@@ -32,3 +32,6 @@ build
 replay_pid*
+
+# Sensitive Files #
+**/application.yml
```
이거없으면 안돌아가는거아니에요? example이라도 올려주세요

anewjean replied:
옙

changchanghwang commented on build.gradle:
```diff
@@ -33,14 +33,19 @@ dependencies {
 //    implementation 'io.jsonwebtoken:jjwt:0.12.3'
     implementation 'com.auth0:java-jwt:4.2.1'
     runtimeOnly 'com.mysql:mysql-connector-j'
+    runtimeOnly 'com.h2database:h2'
```
why need?

anewjean replied:
mysql database 만들기 귀차나서 h2로 테스트해보려고 추가했슴다

changchanghwang replied:
Mysql로해주세요

anewjean replied:
반영했습니다

changchanghwang commented on src/main/kotlin/com/example/antserver/domain/auth/RefreshToken.kt:
```diff
@@ -0,0 +1,57 @@
+@Table(name = "refresh_token")
+data class RefreshToken(
+    @Id
+    @Column(name = "")
```
why no name?

anewjean replied:
; 수정했습니다

changchanghwang commented on src/main/kotlin/com/example/antserver/presentation/UserController.kt:
```diff
@@ -0,0 +1,54 @@
+    fun authenticate(
+        @RequestBody userAuthRequest: UserAuthRequest
+    ): CommonResponse<UserAuthResult> {
+        val authorizationCode = userAuthRequest.authorizationCode
+        val provider = userAuthRequest.provider
+
+        val userAuthResult = userService.authenticateUser(authorizationCode, provider)
```
그냥 request 넘기면 되는데 왜 굳이 분해해서 넘기나요?

anewjean replied:
원래는 controller에서 DTO <-> Entity 변환을 의도한건데 방법을 잘 모르겠어요

changchanghwang commented on src/main/kotlin/com/example/antserver/application/UserService.kt:
```diff
@@ -0,0 +1,42 @@
+        val accessToken = authService.generateAccessToken(user.id)
+        val refreshToken = authService.generateRefreshToken()
+
+        return UserAuthResult(accessToken, refreshToken, user.name, user.email, user.role)
```
왜 return에 accessToken, refreshToken 외에 다른애들이 있는건가요? + 회원가입인지 아닌지 판단하는게 필요합니다.

anewjean replied:
access, refres token만 남겼습니다

anewjean committed:
[WIP] Fix: auth 관련 401 에러 수정

- JWT 사용시 요청시마다 Filter가 작동해 SecurityConfig에서 'users/auth'는 Filter 없이 요청을 수행하도록 수정함

anewjean committed:
Chore: 설정 파일 추가

anewjean committed:
Fix: oauth 관련 변수들 application 저장 변수와 동일하게 수정

anewjean committed:
Chore: oauth관련 변수 모두 application.yml에서 관리하도록 변경

anewjean committed:
[WIP] Fix: auth 구글 소셜로그인 시 URL-encoded Form Data 형식 안 맞는 문제 수정

- TODO: 코틀린에서 제공하는 URL-encoded Form Data 포맷팅 메소드 찾아서 적용

anewjean committed:
Fix: auth 미완 로직 보완 및 버그 수정

- refresh token 발급 추가
- UUID v4 -> v7 변경
- BaseEntity -> AggregateRoot 네이밍 변경
- google token 요청 시 UriComponentsBuilder 사용하도록 변경
- enum type 저장을 위한 @Enumerated(EnumType.STRING) 추가

anewjean committed:
Rename: AggregateRoot 위치 util -> infrastructure 변경

changchanghwang commented on src/main/kotlin/com/example/antserver/domain/auth/RefreshToken.kt:
```diff
@@ -0,0 +1,35 @@
+package com.example.antserver.domain.auth
+
+import com.example.antserver.domain.user.User
```
remove useless import

changchanghwang commented on src/main/kotlin/com/example/antserver/application/UserService.kt:
```diff
@@ -0,0 +1,46 @@
+    // user 삭제
+    fun deleteUser() {
+        TODO("Not yet implemented")
+    }
```
YAGNI

changchanghwang commented on src/main/kotlin/com/example/antserver/application/UserService.kt:
```diff
@@ -0,0 +1,46 @@
+        val accessToken = authService.generateAccessToken(user.id)
+        val refreshToken = authService.generateRefreshToken()
+
+        // refresh token 저장
```
useless comment

changchanghwang commented on src/main/kotlin/com/example/antserver/application/UserService.kt:
```diff
@@ -0,0 +1,46 @@
+        val googleUser = authService.authenticateThroughGoogle(userAuthRequest.authorizationCode)
+        val user = authService.authenticateByEmailOrRegister(googleUser, userAuthRequest.provider)
+
+        // token 발급
```
useless comment

changchanghwang commented on src/main/kotlin/com/example/antserver/application/UserService.kt:
```diff
@@ -0,0 +1,46 @@
+
+    @Transactional
+    fun authenticateUser(userAuthRequest: UserAuthRequest): UserAuthResponse {
+        // 구글 소셜 로그인을 통해 유저 인증
```
useless comment

changchanghwang commented on src/main/kotlin/com/example/antserver/domain/user/User.kt:
```diff
@@ -0,0 +1,69 @@
+
+    fun deleteUser(user: User) {
+        TODO("Not yet implemented")
+    }
```
YAGNI

changchanghwang commented on src/main/kotlin/com/example/antserver/domain/user/UserRepository.kt:
```diff
@@ -0,0 +1,10 @@
+    fun save(user: User): User
+    fun findById(userId: UUID): User?
+    fun findByEmail(email: String): User?
+//    fun findByProviderId(providerId: String): User
```
YAGNI

changchanghwang commented on src/main/kotlin/com/example/antserver/infrastructure/user/JpaUserRepository.kt:
```diff
@@ -0,0 +1,11 @@
+interface JpaUserRepository: JpaRepository<User, UUID> {
+    fun save(user: User): User
+    fun findByEmail(email: String): User?
+//    fun findByProviderId(providerId: String): User
```
YAGNI

changchanghwang commented on src/main/kotlin/com/example/antserver/infrastructure/user/UserRepositoryImpl.kt:
```diff
@@ -0,0 +1,26 @@
+
+//    override fun findByProviderId(providerId: String): User {
+//        TODO("Not yet implemented")
+//    }
```
YAGNI

changchanghwang commented on src/main/kotlin/com/example/antserver/presentation/UserController.kt:
```diff
@@ -0,0 +1,49 @@
+     1. Google 소셜 로그인을 통한 유저 인증
+     2. 기존 유저가 아니라면 신규 등록
+     3. JWT로 Access Token & Refresh Token 발급
+    */
```
why comment here?

changchanghwang commented on src/main/kotlin/com/example/antserver/presentation/UserController.kt:
```diff
@@ -0,0 +1,49 @@
+    // user 삭제 - ADMIN
+    fun deleteUser() {
+        TODO("Not yet implemented")
+    }
```
YAGNI

changchanghwang commented on src/main/kotlin/com/example/antserver/util/config/SecurityConfig.kt:
```diff
@@ -0,0 +1,44 @@
+                .formLogin { it.disable() }
+                .headers { it.frameOptions { frameOptions -> frameOptions.sameOrigin() } }
+                .cors { } // CORS 에러 방지용 TODO.추가
+                .sessionManagement { it.sessionCreationPolicy(SessionCreationPolicy.STATELESS) } // Stateless 세션 설정
```
이건 왜 필요한건가요?

changchanghwang commented on src/main/kotlin/com/example/antserver/util/config/SecurityConfig.kt:
```diff
@@ -0,0 +1,44 @@
+                .httpBasic { it.disable() }
+                .formLogin { it.disable() }
+                .headers { it.frameOptions { frameOptions -> frameOptions.sameOrigin() } }
+                .cors { } // CORS 에러 방지용 TODO.추가
```
코멘트를 이해 못했습니다. 뭘 추가하는건가요?

anewjean replied:
저희 도메인 추후에 추가하려고 합니다. 코멘트 보완했어요

changchanghwang commented on src/main/resources/application.properties:
```diff
@@ -1 +0,0 @@
-spring.application.name=antserver
```
왜 삭제되었나요?

anewjean replied:
application.yml사용해서 properties는 안 씁니다

changchanghwang commented on src/main/kotlin/com/example/antserver/util/response/CommonResponse.kt:
```diff
@@ -0,0 +1,19 @@
+    val data: T,
+) {
+    companion object {
+        fun <T> success(data: T): CommonResponse<T> {
+            return CommonResponse(
+                data = data
+            )
+        }
+
+        fun <String> fail(errorMessage: String): CommonResponse<String> {
+            return CommonResponse(
+                data = errorMessage
+            )
+        }
```
왜 나누는건가요?

애초에 그냥 companion object없이 사용하면 되는거 아닌가요?

anewjean replied:
반영했습니다

changchanghwang commented on src/main/kotlin/com/example/antserver/application/AuthService.kt:
```diff
@@ -0,0 +1,140 @@
+
+@Service
+class AuthService(
+    @Value("\${oauth.google.client-id}") private val clientId: String,
+    @Value("\${oauth.google.client-secret}") private val clientSecret: String,
+    @Value("\${oauth.google.redirect-uri}") private val redirectUri: String,
+    @Value("\${oauth.google.token-url}") private val tokenUrl: String,
+    @Value("\${oauth.google.userinfo-url}") private val userInfoUrl: String,
+    @Value("\${jwt.secret}") val secretKey: String,
+    @Value("\${jwt.expiration-time.access}") val accessTokenExpirationTime: Long,
+    @Value("\${jwt.expiration-time.refresh}") val refreshTokenExpirationTime: Long,
```
이거 환경변수 관리하는 configStore같은거 하나 만들어서 그거 주입받아서 사용하죠. 환경변수는 전부 거기서 관리하고

anewjean replied:
반영했습니다.

changchanghwang commented:
폴더구조가 이상합니다. 어떤건 layer에 그냥 바로 파일들이 나열되어있고 어떤건  layer안에서 domain마다로 나뉘어져있고... 통일 부탁드립니다.

changchanghwang commented on src/main/kotlin/com/example/antserver/application/UserService.kt:
```diff
@@ -0,0 +1,46 @@
+) {
+
+    @Transactional
+    fun authenticateUser(userAuthRequest: UserAuthRequest): UserAuthResponse {
```
얘는 왜 user service인건가요?

anewjean replied:
authService에는 token 관련 로직만 두려고 합니다. UserService에서도 갖다 쓰고, JwtAuthenticationFilter에서도 갖다쓰도록
controller에서도 users/auth 쓰듯이 소셜로그인 로직은 UserService에 두는 게 맞는 것 같아요

changchanghwang replied:
그럼 authService가 아니라 tokenService아닌가요?

anewjean committed:
Refactor: application.yml에서 가져오는 변수처리 @Value -> @ConfigurationProperties로 변경

changchanghwang commented on src/main/kotlin/com/example/antserver/application/AuthService.kt:
```diff
@@ -0,0 +1,140 @@
+            ?: throw IllegalArgumentException("유효하지 않은 Authorization Code입니다.")
+    }
+
+    fun fetchGoogleUser(googleJwtToken: String): GoogleUserResponse {
```
getGoogleProfile or getGoogleUser 는 어떤가요?

anewjean replied:
getGoogleProfile로 변경했습니다

changchanghwang commented on src/main/kotlin/com/example/antserver/application/AuthService.kt:
```diff
@@ -0,0 +1,140 @@
+        return fetchGoogleUser(googleJwtToken)
+    }
+
+    fun fetchGoogleJwtToken(authorizationCode: String): String {
```
getGoogleAccessToken?

anewjean replied:
변경완료

anewjean committed:
Test: AuthService 테스트코드 추가

changchanghwang commented on src/main/kotlin/com/example/antserver/application/AuthService.kt:
```diff
@@ -0,0 +1,140 @@
+            .sign(Algorithm.HMAC512(secretKey))
+    }
+
+    fun upsertRefreshToken(userId: UUID, newRefreshToken: String) {
```
너무 인프라 메서드이름같아요

anewjean replied:
renew로 변경했습니다

changchanghwang commented on src/main/kotlin/com/example/antserver/application/AuthService.kt:
```diff
@@ -0,0 +1,140 @@
+            .verify(accessToken)
+            .getClaim(USER_ID_CLAIM)
+            ?.asString()
+    }.getOrNull()
```
?

anewjean replied:
값 없으면 throw 하도록 변경했습니다

changchanghwang commented on src/main/kotlin/com/example/antserver/application/AuthService.kt:
```diff
@@ -0,0 +1,140 @@
+            ?.trim()
+    }
+
+    fun getRefreshToken(request: HttpServletRequest): String? {
+        return request.cookies
+            ?.find { it.name == "RefreshToken" }?.value
+    }
```
?

anewjean replied:
값 없으면 throw 하도록 변경했습니다

changchanghwang replied:
아뇨 request.cookies가 왜 application에 있나요?

changchanghwang commented on src/main/kotlin/com/example/antserver/application/AuthService.kt:
```diff
@@ -0,0 +1,140 @@
+            ?.takeIf { it.startsWith(BEARER_PREFIX) }
+            ?.removePrefix(BEARER_PREFIX)
+            ?.trim()
+    }
```
애는 왜 서비스단에...?

anewjean replied:
그럼 어디에...?

changchanghwang replied:
presentation layer에 있어야되지 않을까요?

changchanghwang commented on build.gradle:
```diff
@@ -29,11 +29,21 @@ repositories {
+    // database
     runtimeOnly 'com.mysql:mysql-connector-j'
+    // uuid
+    implementation 'com.github.f4b6a3:uuid-creator:5.2.0'
```
https://github.com/cowtowncoder/java-uuid-generator

왜 이런 많이 쓰이는거 안쓰시고 이런걸...

anewjean replied:
변경했습니다

changchanghwang commented on src/main/kotlin/com/example/antserver/application/AuthService.kt:
```diff
@@ -0,0 +1,140 @@
+
+    fun authenticateByEmailOrRegister(googleUser: GoogleUserResponse, provider: ProviderType): User {
+        val email = googleUser.email
+        return userRepository.findByEmail(email) ?: userRepository.save(
+            User.of(
+                name = googleUser.name,
+                email = googleUser.email,
+                provider = provider,
+                providerId = googleUser.sub,
+                role = UserRoleType.MEMBER
+            )
+        )
```
이건 왜 authService?

anewjean replied:
user로 옮겼습니다

changchanghwang commented on src/main/kotlin/com/example/antserver/domain/user/User.kt:
```diff
@@ -0,0 +1,69 @@
+data class User(
+    @Id
+    @Column(name = "id")
+    val id: UUID = UuidCreator.getTimeOrderedEpoch(), // v6
```
wrong comment

changchanghwang commented on src/main/kotlin/com/example/antserver/presentation/dto/user/UserDetailResponse.kt:
YAGNI

changchanghwang commented on src/main/kotlin/com/example/antserver/util/config/SecurityConfig.kt:
```diff
@@ -0,0 +1,44 @@
+
+import com.example.antserver.util.filter.CustomAccessDeniedHandler
+import com.example.antserver.util.filter.CustomAuthenticationEntryPoint
+//import com.example.antserver.util.filter.JwtAuthenticationFilter
```
remove useless import

changchanghwang commented on src/main/kotlin/com/example/antserver/util/config/SecurityConfig.kt:
```diff
@@ -0,0 +1,44 @@
+class SecurityConfig(
+    private val customAuthenticationEntryPoint: CustomAuthenticationEntryPoint,
+    private val customAccessDeniedHandler: CustomAccessDeniedHandler,
+//    private val jwtAuthenticationFilter: JwtAuthenticationFilter,
```
remove

changchanghwang commented on src/main/kotlin/com/example/antserver/util/filter/JwtAuthenticationFilter.kt:
왜 주석처리되어있나요 어차피 쓰는곳도 없는데

anewjean replied:
컴파일 에러..

changchanghwang replied:
쓰는곳 없으면 에러나요..?

anewjean replied:
안 써도 에러나던데요

anewjean committed:
Feat: CustomException 추가 및 AuthService 적용

anewjean committed:
Feat: 유저 이름 변경, 유저 정보 조회 기능 추가

anewjean committed:
Rename: AggregateRoot 위치 infrastructure -> domain 변경

anewjean committed:
Refactor: 불필요한 주석 삭제, 메서드명 변경, 폴더 구조 개선 등

changchanghwang commented on src/main/kotlin/com/example/antserver/application/auth/AuthService.kt:
```diff
@@ -0,0 +1,75 @@
+    private val refreshTokenRepository: RefreshTokenRepository,
+) {
+
+    companion object {
+        private const val BEARER_PREFIX = "Bearer "
+        private const val ACCESS_TOKEN_SUBJECT = "AccessToken"
+        private const val REFRESH_TOKEN_SUBJECT = "RefreshToken"
+        private const val USER_ID_CLAIM = "userId"
+    }
```
얘네는 왜 JwtProperties에 안들어가있는건가요?

anewjean replied:
넣겠습니다

changchanghwang commented on src/main/kotlin/com/example/antserver/application/auth/AuthService.kt:
```diff
@@ -0,0 +1,75 @@
+        return header.takeIf { it.startsWith(BEARER_PREFIX) }
+            ?.removePrefix(BEARER_PREFIX)
+            ?.trim()
+            ?: throw AuthenticationException("Bearer <token> 형식이 맞지 않습니다.")
```
이런 에러메세지들은 서버에서만 나오는건가요 아니면 클라이언트로까지 보내주는건가요?

anewjean replied:
클라이언트까지 보내줍니다

anewjean replied:
"인증 오류입니다"로 덮어써져서 클라이언트한테 나갑니다

changchanghwang commented on src/main/kotlin/com/example/antserver/application/user/UserService.kt:
```diff
@@ -0,0 +1,108 @@
+        val user = authenticateByEmailOrRegister(googleUser, userAuthRequest.provider)
+
+        val accessToken = authService.generateAccessToken(user.id)
+        val refreshToken = authService.generateRefreshToken()
```
비동기로 처리가능한가요? (use coroutine?)

changchanghwang commented on src/main/kotlin/com/example/antserver/application/user/UserService.kt:
```diff
@@ -0,0 +1,108 @@
+    }
+
+    fun getGoogleProfile(googleJwtToken: String): GoogleProfileResponse {
+        return RestTemplate().getForEntity(
```
궁금한게 있는데 왜 httpClient 같은거 안쓰시고 이거 쓰시는건가요?

anewjean replied:
OAuth2 Client 쓰면 편리하다고 하던데 그냥 직접 해보고싶었어요

changchanghwang replied:
https://docs.spring.io/spring-framework/reference/web/webflux-webclient.html

이런거쓰면 조금 더 직관적이지 않나요

anewjean committed:
Fix: 유저 이름 등록, 유저 정보 확인 시 Bearer prefix 삭제 로직 추가 등

+ 유저 정보 확인 시 UserDetailsResponse 필드가 private이어서 응답값 반환 안 되는 버그 수정

anewjean committed:
[WIP] Feat: Poll, Schedule 생성 기능 추가

- Poll Scheduler로 Poll을 생성 -> 이벤트 발행
- PollStatus별 Poll 조회
- Poll 생성 이벤트 발행 시 Schedule 생성
- pollId별 Schedule 리스트 조회
- schedule 투표

anewjean committed:
Chore: 불필요한 import 제거

anewjean committed:
[WIP] Feat: Participation 기능 추가

- Schedule CONFIRMED 이벤트 발생하면 Participation을 영속화함
- useId별 Schedule 조회

anewjean committed:
[WIP] Feat: Participation 기능 추가

- Schedule CONFIRMED 이벤트 발생하면 Participation을 영속화함
- useId별 Schedule 조회

changchanghwang commented on src/main/kotlin/com/example/antserver/domain/AggregateRoot.kt:
```diff
@@ -0,0 +1,24 @@
+    @CreatedDate
+    @Column(nullable = false, updatable = false)
+    var createdAt: Instant? = Instant.now()
+        protected set
```
이건 왜필요한건가요?

anewjean replied:
protected set이요? 이건 테스트해보면서 지웠슴다

changchanghwang replied:
안지워져있는데요..?

anewjean replied:
누락됐었나봐요 방금 다시 반영했습니다

anewjean committed:
[WIP] Fix: 스케줄 확정 시 participation 저장 로직 추가

anewjean committed:
[WIP] Fix: 스케줄 확정 시 participation 저장 로직 추가

anewjean committed:
[WIP] Fix: 스케쥴 확정 로직 추가 등

anewjean committed:
Feat: Access Token 만료 시 재발급 로직 추가

anewjean committed:
Update: 유저 이름 변경 api 스펙 변경 

변경할 name을 path variable -> request body로 가져오도록 수정

anewjean committed:
Create: UserNameRequest

anewjean committed:
Rename: UserNameRequest to UserNameRequest.kt

anewjean committed:
Fix: api 스팩 변경 및 테스트코드 추가

anewjean committed:
Fix: api 스팩 변경 및 테스트코드 추가

anewjean committed:
Fix: api 스팩 변경 및 테스트코드 추가

anewjean committed:
Fix: api 스팩 변경 및 테스트코드 추가

anewjean committed:
Fix: api 스팩 변경 및 테스트코드 추가

anewjean committed:
Feat: JwtAuthenticationFilter 추가 및 user role에 prefix 추가

- OncePerRequestFilter 상속받은 JwtAuthenticationFilter가 클라이언트의 요청에서 access token을 분석해 접근을 관리하도록 함
- Spring에서 User의 Role별로 요청 접근을 관리하기 위해서 ROLE_ prefix가 필요해서 추가함

changchanghwang commented:
![image](https://github.com/user-attachments/assets/29fd3ad8-425a-47d6-ad59-b542c7dd46e6)

changchanghwang commented on src/main/kotlin/com/example/antserver/util/filter/JwtAuthenticationFilter.kt:
```diff
@@ -0,0 +1,64 @@
+        response: HttpServletResponse,
+        filterChain: FilterChain,
+    ) {
+        val excludedPaths = listOf("/users/auth", "/auth/refresh")
```
왜 여기서 exclude하나요? security에서 안하고?

anewjean replied:
SecurityFilterChain에서 JwtAuthenticationFilter를 UsernamePasswordAuthenticationFilter 앞쪽에 배치해놔서 여기서도 exclude 해줘야 /users/auth, auth/refresh가 돌더라고요

changchanghwang replied:
filter는 security에서 permitAll을 해도 동작함. catch해서 expired가 아닐때를 제외하고는 그냥 doFilter로 넘기고 (다음 filter에서 useName어쩌구가 permitAll에 따라 알아서 해줄거임.) expired면 throw exception

changchanghwang commented on src/main/kotlin/com/example/antserver/util/filter/JwtAuthenticationFilter.kt:
```diff
@@ -0,0 +1,64 @@
+        val accessToken = authService.getAccessToken(request).takeIf(authService::isTokenValid)
+            ?: return false
+
+        authService.parseClaims(accessToken)
```
여기서 에러가 나면 true, false를 주는게 아니라 exception이 throw되는거 아닌가요?

anewjean replied:
엇 맞아요

anewjean committed:
Fix: JwtAuthenticationFilter 예외처리 및 오류메세지 버그 수정

anewjean committed:
Fix: h2 메인 페이지도 인증 제외

changchanghwang commented on src/main/kotlin/com/example/antserver/presentation/auth/AuthController.kt:
```diff
@@ -0,0 +1,30 @@
+) {
+
+    @GetMapping("/refresh")
+    fun renewAccessToken(
```
```suggestion
    fun refreshAccessToken(
```

anewjean replied:
반영했습니다

anewjean replied:
근데 저도 refresh 생각했었고 refreshToken도 재발급 메서드 있는데 refreshRefreshToken은 이상한 것 같아서 accessToken이랑 통일하려고 renew로 했던건데 refreshRefreshToken 괜찮나요?

changchanghwang commented on src/main/kotlin/com/example/antserver/presentation/auth/AuthController.kt:
```diff
@@ -0,0 +1,30 @@
+    ): CommonResponse<RefreshResponse> {
+        val currentAccessToken = authService.getAccessToken(request)
+        val userId = UUID.fromString(authService.parseClaimsWithoutVerify(currentAccessToken))
+        val refreshedAccessToken = authService.renewAccessToken(userId, refreshRequest.refreshToken)
```
```suggestion
        val newAccessToken = authService.refreshAccessToken(userId, refreshRequest.refreshToken)
```

anewjean replied:
반영했습니다

anewjean committed:
Refactor: JwtProperties로 관리 가능한 변수 추가

anewjean commented:
> ![image](https://private-user-images.githubusercontent.com/86486778/397172332-29fd3ad8-425a-47d6-ad59-b542c7dd46e6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ2MTQ0MTAsIm5iZiI6MTczNDYxNDExMCwicGF0aCI6Ii84NjQ4Njc3OC8zOTcxNzIzMzItMjlmZDNhZDgtNDI1YS00N2Q2LWFkNTktYjU0MmM3ZGQ0NmU2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjE5VDEzMTUxMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTgwZWNiNzAzYzQwYzNkOGI0YzFjY2YyMDkyNDkwM2QwNmMxNTI4YmI3ZGI4ZDhhNjQ4ZGE0NTYzMmJmMTBkYjImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.PTbKTPecTEK4lq-WQM4g7ZR2E63a3wlWOXD6ZEA9VEw)

<img width="1088" alt="Screenshot 2024-12-19 at 10 33 09 PM" src="https://github.com/user-attachments/assets/82327149-2be1-4699-935d-bb02cc33b4ec" />
엥...

anewjean committed:
Rename: AuthService -> TokenService

anewjean committed:
Fix: 인증 관련 오류메세지는 Access token 만료 이외에 상세 이유를 알 수 없도록 수정

anewjean committed:
Rename: application.auth.TokenService -> util.jwt.JwtTokenManager

anewjean committed:
Fix: protected set 제거

상속받은 클래스 내부에서 상태 변경 불필요

anewjean committed:
Fix: JwtAuthenticationFilter 테스트 가능하도록 private fun -> fun 수정 등

 - JwtTokenManager.renewRefreshToken -> refreshRefreshToken으로 메서드명 변경

anewjean committed:
유저 인증 시 비동기 처리 적용

changchanghwang commented on src/main/kotlin/com/example/antserver/application/user/UserService.kt:
```diff
@@ -0,0 +1,108 @@
+        val user = userRepository.findById(userId)
+            ?: throw UserNotFoundException("유저를 찾을 수 없습니다.")
+        user.updateName(newName)
+        return userRepository.save(user)
```
No return entity.

entity를 걸러낼 수 있는법이 컨트롤러단에 있다면 모르겠지만 이런경우 엔티티 전체를 반환하기 때문에 좋지 않아보입니다. -> 제공되지 않아야할 프로퍼티들이 제공될 수 있음.

anewjean replied:
클라한테는 "${userNameRequest.name}님의 이름이 정상 등록되었습니다." 만 반환해요

changchanghwang replied:
그럼 리턴이 필요한가요?

anewjean replied:
아니요..

anewjean replied:
수정했어요

changchanghwang commented on src/main/kotlin/com/example/antserver/domain/auth/RefreshToken.kt:
```diff
@@ -0,0 +1,34 @@
+    var token: String,
+): AggregateRoot() {
+
+    companion object {
+        fun of(userId: UUID, token: String): RefreshToken {
+            return RefreshToken(
+                userId = userId,
+                token = token
+            )
+        }
+    }
```
why using factory method?

anewjean replied:
무지성 companion object ........ "이 값을 기반으로 인스턴스 생성한다"는 의미를 전달하려고 of 쓰는 거 아닌가요

changchanghwang replied:
Constructor로 생성하는것과 별 차이가 없어보이긴 합니다

changchanghwang commented on src/main/kotlin/com/example/antserver/presentation/auth/AuthController.kt:
```diff
@@ -0,0 +1,30 @@
+        request: HttpServletRequest,
+        @RequestBody refreshRequest: RefreshRequest
+    ): CommonResponse<RefreshResponse> {
+        val currentAccessToken = jwtTokenManager.getAccessToken(request)
+        val userId = UUID.fromString(jwtTokenManager.parseClaimsWithoutVerify(currentAccessToken))
+        val newAccessToken = jwtTokenManager.refreshAccessToken(userId, refreshRequest.refreshToken)
```
why need accessToken?

anewjean replied:
access token 발급 로직 자체가 userId를 클레임으로 거는거라 재발급하려면 기존 access token에서 기존 userId를 parsing해야해요

changchanghwang replied:
RefreshToken에 userId 저장하고있는거아닌가요

anewjean replied:
🫠 후 맞아요 수정했어요

changchanghwang commented on src/main/kotlin/com/example/antserver/presentation/auth/dto/RefreshResponse.kt:
```diff
@@ -0,0 +1,11 @@
+data class RefreshResponse(
+    val accessToken: String
+) {
+    companion object {
+        fun of(accessToken: String): RefreshResponse {
+            return RefreshResponse(accessToken)
+        }
+    }
```
why need?

changchanghwang commented on src/main/kotlin/com/example/antserver/util/jwt/JwtTokenManager.kt:
```diff
@@ -0,0 +1,84 @@
+            ?: throw AuthenticationException("Access Token에서 userId 클레임을 추출할 수 없습니다.")
+    }
+
+    fun getAccessToken(request: HttpServletRequest): String {
+        val header = request.getHeader(HttpHeaders.AUTHORIZATION)
+            ?: throw AuthenticationException("Authorization 헤더가 없습니다.")
+        return header.takeIf { it.startsWith(jwtProperties.bearerPrefix) }
+            ?.removePrefix(jwtProperties.bearerPrefix)
+            ?.trim()
+            ?: throw AuthenticationException("Bearer <token> 형식이 맞지 않습니다.")
+    }
```
왜필요한가요?

anewjean replied:
개발 단계에서 필요했어요

anewjean replied:
지우겠습니다

changchanghwang commented on src/main/kotlin/com/example/antserver/util/config/SecurityConfig.kt:
```diff
@@ -0,0 +1,44 @@
+            .sessionManagement { it.sessionCreationPolicy(SessionCreationPolicy.STATELESS) } // Stateless 세션 설정
+            .authorizeHttpRequests { auth ->
+                auth.requestMatchers(HttpMethod.OPTIONS).permitAll() // CORS Preflight 방지
+                auth.requestMatchers("/h2-console", "/h2-console/**", "/health", "/users/auth", "/auth/refresh").permitAll()
```
remove useless paths

anewjean replied:
h2요?

changchanghwang commented on src/main/kotlin/com/example/antserver/util/exception/CustomAuthenticationEntryPoint.kt:
```diff
@@ -0,0 +1,30 @@
+@Component
+class CustomAuthenticationEntryPoint: AuthenticationEntryPoint {
+    override fun commence(
+        request: HttpServletRequest?,
+        response: HttpServletResponse?,
+        authException: AuthenticationException?
```
실제로 nullable한가요?

anewjean replied:
아니네요;

changchanghwang commented on src/main/kotlin/com/example/antserver/util/jwt/JwtAuthenticationFilter.kt:
```diff
@@ -0,0 +1,67 @@
+        return try {
+            jwtTokenManager.getAccessToken(request).takeIf(jwtTokenManager::isTokenValid)
+                ?: throw AuthenticationException("Access token이 만료되었습니다.")
+        } catch (ex: AuthenticationException) {
+            request.setAttribute("customAuthErrorMessage", ex.message)
+            throw ex
+        }
```
이정도는 별로 상관없지만 개인적인 경험으로는 웬만하면 단축어는 사용하지 않는게 좋습니다.

anewjean replied:
옙

changchanghwang commented on src/main/kotlin/com/example/antserver/util/exception/CustomAuthenticationEntryPoint.kt:
```diff
@@ -0,0 +1,30 @@
+                    "errorMessage": "$errorMessage"
+                }
+            }
+        """.trimIndent())
```
이건 뭔가요?

anewjean replied:
""" """ 안의 들여쓰기 제거해줍니다. 들여쓰고 trimIndent 안 쓰면 되는데 그냥 써봤어요

changchanghwang replied:
제가 이해를 잘 못했는데 예시하나만 부탁드립니다

anewjean replied:
"""
            {
                "data": {
                    "errorMessage": "$errorMessage"
                }
            }
        """
        여기에 trimIndent를 적용하면
        
        """
        {
            "data": {
                "errorMessage": "$errorMessage"
            }
        }
        """
        이렇게 알아서 인덴트를 맞춰줍니다

anewjean committed:
Feat: logger 추가 및 서버용, 클라이언트용 에러메세지 분리

- 클라이언트용 AuthenticationException 에러는 '인증 오류입니다.'로 통일

anewjean committed:
Modify: 유저 인증 시 access token, refresh token 발급 로직 비동기 처리 추가

anewjean committed:
Fix: 구글 인증 로직 비동기 처리 취소

- 유저 인증 시 구글 인증 로직까지 비동기처리 하고 있었는데 동기 처리로 원상 복구함
- cpu bound 작업인 JWT 생성 로직만 비동기 처리

changchanghwang commented on src/main/kotlin/com/example/antserver/domain/user/ProviderType.kt:
```diff
@@ -0,0 +1,5 @@
+package com.example.antserver.domain.user
+
+enum class ProviderType {
+    GOOGLE,
+}
```
얘는 왜 파일로 나뉘어져있나요?

anewjean replied:
리팩토링할 때 ProviderType별로 처리하려고요

changchanghwang replied:
그게 파일로 나뉘어져있는것과 상관이있나요?

anewjean committed:
Refactor: Exception 관련 파라미터 Not nullable하게 변경

anewjean committed:
Fix: cors 비활성화

anewjean committed:
Fix: cors 비활성화

anewjean committed:
Chore: 로그레벨 DEBUG -> INFO 변경

anewjean committed:
Fix: 유저 인증 시 신규 회원인지, 기존 회원인지 알 수 있는 isNew 추가 반환

changchanghwang committed:
fix: remove useless paths, refactor jwtFilter filter logic

changchanghwang committed:
Merge branch 'auth' of https://github.com/TeamOfAnts/Ant-server into auth

changchanghwang committed:
Merge branch 'auth' of https://github.com/TeamOfAnts/Ant-server into poll

changchanghwang committed:
Merge branch 'poll' of https://github.com/TeamOfAnts/Ant-server into participation

changchanghwang committed:
fix: conflict

changchanghwang committed:
fix: conflict

anewjean committed:
Refactor: 모든 ApplicationException은 GlobalExceptionHandler가 처리하도록 수정

anewjean committed:
Refactor: 모든 ApplicationException은 GlobalExceptionHandler가 처리하도록 수정

anewjean committed:
Refactor: 모든 ApplicationException은 GlobalExceptionHandler가 처리하도록 수정

anewjean committed:
Fix: 토큰 검증, 생성, 저장 로직은 TokenService가 처리하도록 변경

anewjean committed:
Fix: 토큰 검증, 생성, 저장 로직은 TokenService가 처리하도록 변경

anewjean committed:
Fix: 토큰 검증, 생성, 저장 로직은 TokenService가 처리하도록 변경

anewjean committed:
Fix: refresh token 발급 시 불필요한 로직 삭제

anewjean committed:
Fix: refresh token 발급 시 불필요한 로직 삭제

anewjean committed:
Fix: refresh token 발급 시 불필요한 로직 삭제

anewjean committed:
Fix: 유저 업데이트 시 유저 엔티티 반환하지 않도록 수정

anewjean committed:
Fix: 유저 업데이트 시 유저 엔티티 반환하지 않도록 수정

anewjean committed:
Fix: 유저 업데이트 시 유저 엔티티 반환하지 않도록 수정

anewjean committed:
Refactor: 분리되어있던 ApplicationException, Exception 처리 메서드 병합

sq

anewjean committed:
Refactor: 분리되어있던 ApplicationException, Exception 처리 메서드 병합

sq

anewjean committed:
Refactor: 분리되어있던 ApplicationException, Exception 처리 메서드 병합

sq

anewjean committed:
Fix: 토큰 유효성에 대한 예외 처리 추가, JwtAuthenticationFilter 제외 path 추가

- JWT에 대한 verify는 TokenExpiredException, JWTDecodeException, SignatureVerificationException, AlgorithmMismatchException, InvalidClaimException 등 다양하지만 비즈니스 로직 상 TokenExpiredException와 그 외로 구분함
- JwtAuthenticationFilter에서 access token이 없는 경우 access token이 없어도 정상인 path (/health, /users/auth, /auth/refresh)는 Finally 통해 dofilter 후 permitAll() 되기를 기대했지만, permitAll 되기 전 에러가 발생하여 JwtAuthenticationFilter 자체를 통과하도록 수정함

anewjean committed:
Fix: 토큰 유효성에 대한 예외 처리 추가, JwtAuthenticationFilter 제외 path 추가

- JWT에 대한 verify는 TokenExpiredException, JWTDecodeException, SignatureVerificationException, AlgorithmMismatchException, InvalidClaimException 등 다양하지만 비즈니스 로직 상 TokenExpiredException와 그 외로 구분함
- JwtAuthenticationFilter에서 access token이 없는 경우 access token이 없어도 정상인 path (/health, /users/auth, /auth/refresh)는 Finally 통해 dofilter 후 permitAll() 되기를 기대했지만, permitAll 되기 전 에러가 발생하여 JwtAuthenticationFilter 자체를 통과하도록 수정함

anewjean committed:
Fix: 토큰 유효성에 대한 예외 처리 추가, JwtAuthenticationFilter 제외 path 추가

- JWT에 대한 verify는 TokenExpiredException, JWTDecodeException, SignatureVerificationException, AlgorithmMismatchException, InvalidClaimException 등 다양하지만 비즈니스 로직 상 TokenExpiredException와 그 외로 구분함
- JwtAuthenticationFilter에서 access token이 없는 경우 access token이 없어도 정상인 path (/health, /users/auth, /auth/refresh)는 Finally 통해 dofilter 후 permitAll() 되기를 기대했지만, permitAll 되기 전 에러가 발생하여 JwtAuthenticationFilter 자체를 통과하도록 수정함

anewjean committed:
Fix: import erorr

anewjean committed:
Fix: import erorr

anewjean committed:
Fix: 삭제한 custom 예외클래스 관련 Expired 코드 수정

anewjean committed:
Fix: 삭제한 custom 예외클래스 관련 Expired 코드 수정

anewjean committed:
Fix: base 브랜치 pull 관련 error 수정

anewjean committed:
Fix: base 브랜치 pull 관련 error 수정

anewjean committed:
Fix: @Scheduled 실행을 위한 @EnableScheduling 추가 등

anewjean committed:
Fix: @Scheduled 실행을 위한 @EnableScheduling 추가 등

anewjean committed:
Test: ScheduleScheduler 동작 시 updateScheduleStatus 실행 확인

anewjean committed:
Test: ScheduleScheduler 동작 시 updateScheduleStatus 실행 확인

anewjean committed:
Fix: auth 변경사항 반영

- JwtTokenManager
- TokenService

anewjean committed:
Fix: auth 변경사항 반영

- JwtTokenManager
- TokenService

anewjean committed:
Fix: Poll 조회 시 없으면 null 반환하도록 수정

anewjean committed:
Fix: Poll 조회 시 없으면 null 반환하도록 수정

anewjean committed:
Fix: 불필요한 import 삭제

anewjean committed:
Fix: 불필요한 import 삭제

anewjean committed:
Chore: 자신의 정보 조회 응답에서 providerId 제외

anewjean committed:
Chore: 자신의 정보 조회 응답에서 providerId 제외

anewjean committed:
Merge branch 'participation' of https://github.com/TeamOfAnts/Ant-server into participation

anewjean committed:
Fix: Access token 만료 시 에러메세지 처리 수정

- Access Token 만료로 인해서 AuthenticationException 발생 시 에러메세지 return
- '/health', '/users/auth', '/auth/refresh'의 경우에도 access token이 만료되면 return하기 때문에 AuthenticationException 커스텀 에러를 만듦
- customAuthenticationEntrypoint 에 잡히는 에러는 모두 에러메세지 '인증 오류입니다.'로 처리

anewjean committed:
Fix: conflict

anewjean committed:
Merge branch 'poll' into participation

anewjean committed:
Fix: access token 없이 허용된 요청에 대한 처리

- /health, /users/auth, /auth/refresh 요청은 access token이 없어도 접근이 허용되어야 하는데 access token이 없어서 NPE가 발생하면 filter.doFilter 실행 이전에 filter chain 밖으로 에러가 전파되어 요청이 처리되지 않음
- 따라서 NPE를 catch하여 doFilter하도록 수정함

anewjean committed:
Refactor: filter chain 내부에서 error 발생 시 응답 생성 함수 공용화

anewjean committed:
Refactor: Oauth 추상화

- user 엔티티의 unique key: email -> email + provider로 변경

anewjean committed:
Rename: util 폴더구조 변경

anewjean committed:
Fix: ApplicationException의 경우 serverErrorMessage를 로깅

anewjean committed:
Refactor: RestTemplate -> Webclient

anewjean committed:
Merge branch 'auth' into poll

anewjean committed:
Fix: poll 종료 시 poll status update로직 추가

anewjean committed:
Fix: poll.endAt을 스케쥴이 끝나는 날짜 -> 투표가 끝나는 날짜로 변경

anewjean committed:
Fix: Schedule에 대한 voter 추가 시 중복 투표 방지로직 추가

anewjean committed:
Refactor: 투표 시 응답 메세지 수정

anewjean committed:
Merge branch 'poll' into participation

anewjean committed:
Fix: import error

anewjean committed:
Fix: 스케쥴 확정시 CONFIRMED 상태의 스케쥴만 Participation 객체로 저장하도록 수정

anewjean committed:
[WIP] WebClient 적용 시 APPLICATION_FORM_URLENCODED 미적용 이슈로 임시적 RestTemplate 사용

anewjean committed:
[WIP] WebClient 적용 시 APPLICATION_FORM_URLENCODED 미적용 이슈로 임시적 RestTemplate 사용

anewjean committed:
Merge branch 'auth' into poll

anewjean committed:
Fix: access token이 유효하지 않아도 filter chain 내부에서 허용 여부 판단할 수 있도록 수정

anewjean committed:
Fix: access token이 유효하지 않아도 filter chain 내부에서 허용 여부 판단할 수 있도록 수정

anewjean committed:
Fix: 모든 요청에 대해서 access token이 유효하지 않은 경우에 대한 처리 추가

- access token이 유효하지 않아도 허용된 요청인 /auth/refresh, /users/auth, /health에 대해서는 dofilter할 수 있도록 수정

anewjean committed:
Fix: Conflict

anewjean committed:
Fix: http GET 표준에 맞지 않는 요청 수정

- auth/refresh, /schedule은 GET method를 사용하면서 Body를 전달하고 있어 http 표준에 적합하도록 POST method로 변경

anewjean committed:
Merge branch 'poll' into auth

anewjean committed:
Fix: 스케쥴 조회 http method POST -> GET

- ScheduleRequest 클래스 삭제

anewjean committed:
Fix: Poll, Schdule response UTC -> KTS

anewjean committed:
Rename: PollScheduler 위치 변경

anewjean committed:
Feat: poll 조회 시 유저가 투표한 scheduleId 리스트 추가 응답

anewjean committed:
Fix: 투표 로직 수정

- 기존: 최초 투표에서 변경 불가능
- 변경: 투표 기간 내에 기존의 투표 이력과 새로운 투표 이력을 비교
  -> 기존 투표 이력에 있는데 새 투표 이력에 없으면 삭제, 기존 투표 이력에 없는데 새 투표 이력에 있으면 추가

anewjean committed:
Fix: 투표 응답에 투표자 명단 추가

anewjean committed:
Fix: ScheduleResponse.votes 삭제

- voters.length로 대체

anewjean committed:
Fix: database schema migration

- Participation.type VARCHAR로 매핑할 수 있도록 @Enumerate(EnumType.String) 어노테이션 추가
- application.yml에서 JPA가 ddl 자동 생성하지 않고 유효성 검증만 하도록 update -> validate로 변경

anewjean committed:
Fix: 투표 기간 변경: 일-화 -> 목-토

anewjean committed:
Feat: 투표 미참여 항목 추가

anewjean committed:
Test: 투표 미참여항목 추가 관련 테스트 수정 및 버그 수정

anewjean committed:
Refactor: 존재하지 않는 스케쥴 id에 투표 할 경우에 대한 예외 처리 추가

anewjean committed:
Remove: 불필요한 파일 삭제

anewjean committed:
Chore: application.yml

- 데이터베이스 관련 설정 삭제
- @Scheduled 비활성화 설정 추가

anewjean committed:
Refactor: 비즈니스 로직을 서비스 -> 도메인으로 위임

anewjean committed:
Refactor: 서비스의 단일책임 원칙을 위해 EventListener 분리

anewjean committed:
Refactor: 서비스의 단일책임 원칙을 위해 EventListener 분리

anewjean committed:
Rename: 이벤트 관련 클래스 위치 도메인 -> 서비스로 변경

anewjean committed:
Feat: scheduleId로 Schedule 조회 기능 추가

anewjean committed:
Feat: scheduleId로 Schedule 조회 기능 추가

anewjean committed:
Chore: 불필요한 코드 삭제

sq

anewjean committed:
Chore: 불필요한 코드 삭제

anewjean committed:
Test: Repository Mocking 및 도메인 테스트 추가

anewjean committed:
Test: Repository Mocking 및 도메인 테스트 추가

anewjean committed:
Test: FakeRepository 추가

- Bean 등록

anewjean committed:
Test: FakeRepository 추가

- Bean 등록

anewjean committed:
Chore: application.yml 중요정보 Github Actions 변수로 등록

anewjean committed:
Chore: sonar.yml 환경변수 위치 변경

anewjean committed:
Chore: .env 파일로 환경변수 관리

anewjean committed:
Chore: @SpringBootTest webEnvironment 명시

anewjean committed:
TokenServiceTest 주석처리

anewjean committed:
Chore: sonar.yml test 추가

anewjean committed:
Chore: 테스트코드에서 DB 설정 무효화

anewjean committed:
Chore: application.yml DB 커넥션 비활성화

anewjean committed:
Chore: 테스트 db로 H2 설정

anewjean committed:
Chore: 테스트코드에서 DB 비활성화 설정 삭제

anewjean committed:
Chore: ApplicationContext 로딩 문제 관련 @DirtiesContext 설정

anewjean committed:
Chore: application-test.yml DB 연결 비활성화

anewjean committed:
Feat: 주말 Schedule은 낮/밤 옵션 추가

- ScheduleOn.Scheduled 타입을 Instant?에서 String?으로 변경
- ScheduleOn.Scheduled의 필드를 schedule(Instant)에서 date(LocalDate), dayOfWeek(DayOfWeek), timeOfDay(String?)으로 변경
- Schedule.scheduleOn은 ScheduleOn.Scheduled의 메소드 toString을 통해 '0000-00-00 (요일) 낮'형태로 저장
- dayOfWeek을 한국 요일로 매핑하는 함수 추가

anewjean committed:
Test: ScheduleOn.Scheduled 변경사항 반영 및 에러 수정

anewjean committed:
Refactor: 스케쥴러 시간 컨트롤이 가능하도록 Clock 주입

anewjean committed:
Chore: 운영코드 vs 테스트코드 빈 충돌에 대해 testconfig @Primary 적용

anewjean committed:
Test: PollScheduler

anewjean committed:
Test: 빈 충돌 관련 에러 수정

anewjean committed:
Refactor: PollService, PollScheduler 테스터블하게 변경

- 외부에서 파라미터로 메소드 변수 주입

anewjean committed:
Chore: sonar.yml test 추가 원복

anewjean committed:
Chore: Hibernate, JPA 비활성화

anewjean committed:
Chore: 빌드시 테스트 비활성화

anewjean committed:
Conflict 수정

anewjean committed:
Conflict 수정

anewjean committed:
Fix: Schedule 속성 변경

- 기존
  - scheduleOn(ScheduleOn): ScheudleOn.Scheduled / ScheduleOn.Unscheduled
   e.g) '날짜 + 요일 + 시간대(주말만)' / '미참여'

- 수정
  : description 필드를 추가하여 기존 scheduleOn 내용을 description으로 사용하고, scheduleOn에는 날짜 정보만 담음
   - scheduleOn(ScheduleOn): ScheduleOn.Scheduled / ScheduleOn.Unscheduled
    e.g) '날짜' / 'null'
   - description(ScheduleDescription): ScheduleDescription.Scheduled / ScheduleDescription.Unscheduled
    e.g) '날짜 + 요일 + 시간대(주말만)' / '미참여'

anewjean committed:
Merge branch 'poll' into auth

anewjean committed:
Chore: build시 테스트 비활성화 -> 활성화 원복, JpaRepositoriesAutoConfiguration 비활성화

anewjean committed:
Chore: 테스트시 웹환경 비활성화

anewjean committed:
Chore: Hibernate 관련 설정 제거

anewjean committed:
Chore: build시 테스트 환경에서 application-test.yml 사용하도록 변경

anewjean committed:
Chore: build시 test 안 하도록 설정
